py/server/deephaven/_udf.py

```diff
@@ -354,112 +354,79 @@ def _parse_signature(fn: Callable) -> _ParsedSignature:
         return p_sig
 
 
-def _is_from_np_type(param_types: Set[type], np_type_char: str) -> bool:
-    """ Determine if the given numpy type char comes for a numpy type in the given set of parameter type annotations"""
-    for t in param_types:
-        if issubclass(t, np.generic) and np.dtype(t).char == np_type_char:
-            return True
-    return False
-
-
-def _convert_arg(param: _ParsedParam, arg: Any) -> Any:
-    """ Convert a single argument to the type specified by the annotation """
-    if arg is None:
-        if not param.none_allowed:
-            raise TypeError(f"Argument {param.name!r}: {arg} is not compatible with annotation {param.orig_types}")
+def _prepare_auto_arg_conv(p_sig: _ParsedSignature, encoded_arg_types: str) -> bool:
+    """ Determine whether the auto argument conversion should be used and set the converter functions for the
+    parameters."""
+    arg_conv_needed = True
+    arg_type_strs = encoded_arg_types.split(",")
```

Comment:
Since we don't yet capture varargs and keyword parameters, we assume that the args provided are valid for the UDF. We do check when a UDF has 0 parameters but is called with some arguments.